File: src/main/java/com/akagiyui/drive/model/response/AnnouncementResponse.java
package com.akagiyui.drive.model.response;

import com.akagiyui.drive.entity.Announcement;
import lombok.Data;
import lombok.experimental.Accessors;

import java.util.Date;
import java.util.List;

/**
 * 公告信息 响应
 *
 * @author AkagiYui
 */
@Data
@Accessors(chain = true)
public class AnnouncementResponse {

    /**
     * 公告id
     */
    private String id;

    /**
     * 标题
     */
    private String title;

    /**
     * 内容
     */
    private String content;

    /**
     * 用户ID
     */
    private String userId;

    /**
     * 已启用
     */
    private boolean enabled;

    /**
     * 发布时间
     */
    private Date createTime;

    /**
     * 修改时间
     */
    private Date updateTime;

    public AnnouncementResponse(Announcement announcement) {
        this.id = announcement.getId();
        this.title = announcement.getTitle();
        this.content = announcement.getContent();
        this.userId = announcement.getAuthor().getId();
        this.enabled = announcement.getEnabled();
        this.createTime = announcement.getCreateTime();
        this.updateTime = announcement.getUpdateTime();
    }

    /**
     * 从公告实体列表转换
     */
    public static List<AnnouncementResponse> fromAnnouncementList(List<Announcement> announcementList) {
        return announcementList.stream()
                .map(AnnouncementResponse::new)
                .toList();
    }
}

File: app/src/main/kotlin/com/akagiyui/drive/model/response/AnnouncementResponse.kt
package com.akagiyui.drive.model.response

import com.akagiyui.drive.entity.Announcement

/**
 * 公告信息 响应
 *
 * @author AkagiYui
 */
class AnnouncementResponse() {

    /**
     * 公告ID
     */
    var id: String? = null

    /**
     * 标题
     */
    var title: String? = null

    /**
     * 内容
     */
    var content: String? = null

    /**
     * 用户ID
     */
    var userId: String? = null

    /**
     * 用户名
     */
    var username: String? = null

    /**
     * 已启用
     */
    var enabled = false

    /**
     * 发布时间
     */
    var createTime: Long = 0

    /**
     * 修改时间
     */
    var updateTime: Long = 0

    constructor(announcement: Announcement) : this() {
        this.id = announcement.id
        this.title = announcement.title
        this.content = announcement.content
        this.userId = announcement.author.id
        this.username = announcement.author.username
        this.enabled = announcement.enabled
        this.createTime = announcement.createTime.time
        this.updateTime = announcement.updateTime.time
    }

    companion object {
        /**
         * 从公告实体列表转换
         */
        fun fromAnnouncementList(announcementList: List<Announcement>): List<AnnouncementResponse> {
            return announcementList.stream()
                .map { announcement -> AnnouncementResponse(announcement) }
                .toList()
        }
    }
}
